pkg/netpol/eval/internal/k8s/netpol.go

```diff
@@ -519,54 +536,39 @@ func (np *NetworkPolicy) scanEgressRules() ([]SingleRuleLabels, error) {
 // else: returns selectors of non-general rules (rules that are not exposed to entire world or cluster).
 // this func assumes rules are legal (rules correctness check occurs later)
 func (np *NetworkPolicy) doRulesExposeToAllDestOrEntireCluster(rules []netv1.NetworkPolicyPeer, rulePorts []netv1.NetworkPolicyPort,
-	isIngress bool) (rulesSelectors []SingleRuleLabels, err error) {
+	isIngress bool) (rulesSelectors []SingleRuleSelectors, err error) {
 	if len(rules) == 0 {
 		err = np.updateNetworkPolicyGeneralConn(true, true, rulePorts, isIngress)
 		return nil, err
 	}
 	for i := range rules {
-		var ruleSel SingleRuleLabels
+		var ruleSel SingleRuleSelectors
 		if rules[i].IPBlock != nil {
 			continue
 		}
-		// note: LabelSelectorAsMap returns a nil map for nil selector,
-		// and returns empty map (map[string]string{}) for empty selector (all)
-		nsSelectorMap, err := metav1.LabelSelectorAsMap(rules[i].NamespaceSelector)
-		if err != nil {
-			return nil, err
-		}
-		podSelectorMap, err := metav1.LabelSelectorAsMap(rules[i].PodSelector)
-		if err != nil {
-			return nil, err
-		}
 		// a rule is exposed to entire cluster if :
-		// 1. the podSelector is nil (no podselector) but the namespaceSelector is empty ({})
+		// 1. the podSelector is nil (no podselector) but the namespaceSelector is empty ({}) not nil
 		// 2. both podSelector and namespaceSelector are empty ({})
-		if (rules[i].PodSelector == nil && isEmptyMap(nsSelectorMap)) ||
-			(isEmptyMap(nsSelectorMap) && isEmptyMap(podSelectorMap)) {
+		if rules[i].NamespaceSelector != nil && rules[i].NamespaceSelector.Size() == 0 &&
+			(rules[i].PodSelector == nil || rules[i].PodSelector.Size() == 0) {
 			err = np.updateNetworkPolicyGeneralConn(false, true, rulePorts, isIngress)
 			return nil, err
 		}
-		// else (selector not empty)
-		ruleSel.PodLabels = podSelectorMap
-		// special case: ns selector is nil but podSelector is not, so the namespace of the rule is
-		// the policy's namespace; adding the k8s namespace name key to ruleSel.NsLabels
-		if rules[i].NamespaceSelector == nil && rules[i].PodSelector != nil {
+		// else selectors' combination specifies end-points (at least one is not nil)
+		ruleSel.PodSelector = rules[i].PodSelector
+		ruleSel.NsSelector = rules[i].NamespaceSelector
+		if rules[i].NamespaceSelector == nil {
+			// special case: ns selector is nil but podSelector is not, so the namespace of the rule is
+			// the policy's namespace; (turn on the indicator)
 			ruleSel.PolicyNsFlag = true
-			ruleSel.NsLabels = map[string]string{common.K8sNsNameLabelKey: np.Namespace}
-		} else {
-			ruleSel.NsLabels = nsSelectorMap
+			nsNameLabelSel := map[string]string{common.K8sNsNameLabelKey: np.Namespace}
```

Comment:
can avoid duplicating the map of single label containing `K8sNsNameLabelKey` ? there is another function building this map currently